src/draw/lv_draw_buf.h

```diff
@@ -247,7 +350,41 @@ static inline void lv_draw_buf_from_image(lv_draw_buf_t * buf, const lv_image_ds
 
 static inline void lv_draw_buf_to_image(const lv_draw_buf_t * buf, lv_image_dsc_t * img)
 {
-    lv_memcpy(img, buf, sizeof(lv_image_dsc_t));
+    lv_memcpy((void *)img, buf, sizeof(lv_image_dsc_t));
+}
+
+/**
+ * Set the palette color of an indexed image. Valid only for `LV_COLOR_FORMAT_I1/2/4/8`
+ * @param draw_buf pointer to an image descriptor
+ * @param id the palette color to set:
+ *   - for `LV_COLOR_FORMAT_I1`: 0..1
+ *   - for `LV_COLOR_FORMAT_I2`: 0..3
+ *   - for `LV_COLOR_FORMAT_I4`: 0..15
+ *   - for `LV_COLOR_FORMAT_I8`: 0..255
+ * @param c the color to set in lv_color32_t format
+ */
+void lv_draw_buf_set_palette(lv_draw_buf_t * draw_buf, uint8_t id, lv_color32_t c);
+
+static inline void lv_image_buf_set_palette(lv_image_dsc_t * dsc, uint8_t id, lv_color32_t c)
+{
+    LV_LOG_WARN("Legacy API, use lv_draw_buf_set_palette instead.");
```

Comment:
I suggest moving all legacy API `lv_api_map.h`.

Maybe deprecated is a better word for it.

Reply:
Yes, the word "deprecated" is better.